modules/packages/LinearAlgebra.chpl

```diff
@@ -2490,6 +2490,151 @@ proc cosm(A: []) throws {
   }
 }
 
+proc sum(A: [?D], param axis=0) where !isCSDom(D) {
+  if axis == 0 then
+    // axis=0 sums along column
+    return forall j in D.dim(1-axis) do (+ reduce abs(A[D.dim(axis), j..j]));
+  else if axis == 1 then
+    // axis=1 sums along row
+    return forall i in D.dim(1-axis) do (+ reduce abs(A[i..i, D.dim(axis)]));
+}
+
+proc sum(A: [?ADom], param axis=0) where isCSDom(ADom) {
+  var (n,t) = ADom.shape;
+  var sps: sparse subdomain(ADom.parentDom);
+  sps += ADom;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in sps {
+      b[j] += A[i,j];
+    }
+    return b;
+  }
+  else if axis == 1 {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall (i,j) in sps {
+      b[i] += A[i,j];
+    }
+    return b;
+  }
+}
+
+
+proc ones(Dom: domain(2), type eltType=real) {
+  const (m,n) = Dom.shape;
+  var A: [Dom] eltType;
```

Comment:
Can we simplify this to 
`var A: [Dom] eltType = 1 : eltType`?

Reply:
Oh wow, Nice! So `1 : eltType` is like typecasting 1 to eltType ?